front/app/containers/Admin/projects/project/analysis/InputsList/SummarizeButton.tsx

```diff
@@ -0,0 +1,67 @@
+import { Box, Button } from '@citizenlab/cl2-component-library';
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import useAnalysisFilterParams from '../hooks/useAnalysisFilterParams';
+import useAddAnalysisSummary from 'api/analysis_summaries/useAddAnalysisSummary';
+import useAddAnalysisSummaryPreCheck from 'api/analysis_summary_pre_check/useAddAnalysisSummaryPreCheck';
+import { ISummaryPreCheck } from 'api/analysis_summary_pre_check/types';
+
+type Props = {
+  inputsCount?: number;
+};
+const SummarizeButton = ({ inputsCount }: Props) => {
+  const { mutate: addSummary, isLoading: isLoadingSummary } =
+    useAddAnalysisSummary();
+  const { mutate: addSummaryPreCheck, isLoading: isLoadingPreCheck } =
+    useAddAnalysisSummaryPreCheck();
+  const { analysisId } = useParams() as { analysisId: string };
+  const filters = useAnalysisFilterParams();
+  const handleSummaryCreate = () => {
+    addSummary({
+      analysisId,
+      filters,
+    });
+  };
+
+  const [preCheck, setPreCheck] = useState<ISummaryPreCheck | null>(null);
+  const jsonFilters = JSON.stringify(filters);
```

Comment:
I wonder if we shouldn't try to stabilize the filters reference at the level of `useAnalysisFilterParams` hook, instead of using this trick with stringifying it. Otherwise we have to do this every time we want to use it in a useEffect (which, to be fair, I don't know that it will be this often).

Reply:
Good point, did that now (by using the same trick) in the hook. Is there a nicer way of doing this, or is that fine?

Reply:
Usually we use useMemo for values and useCallback for methods. Would that work in this case?

Reply:
Yes, that's what I did, but then I still need a way to to get these filters in the dependencies of useMemo, so I used serializing to json for that for it to be stable. And I'm not sure whether that's "the right way", though I'm sure it works :)

Reply:
It does work, as long as we make sure everything is serialisable which, in this case, I think it is so we shouldn't have an issue. 👍